core/src/test/java/org/apache/iceberg/TestRowDelta.java

```diff
@@ -1270,11 +1332,16 @@ public void testConcurrentNonConflictingRowDeltaAndRewriteFilesWithSequenceNumbe
                 baseSnapshot.sequenceNumber())
             .validateFromSnapshot(baseSnapshot.snapshotId());
 
-    rowDelta.commit();
-    rewriteFiles.commit();
+    commit(table, rowDelta, branch);
+    commit(table, rewriteFiles, branch);
 
-    validateTableDeleteFiles(table, deleteFile1);
-    validateTableFiles(table, dataFile2);
+    if (branch == "testBranch") {
```

Comment:
equals() instead of ==

Reply:
Also just realized we don't need to differentiate the two cases here, I think we could just have a single method which takes in a branch (and the branch could just be main or not, it doesn't really matter)